src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestCreationParameters.cs

```diff
@@ -0,0 +1,129 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Protocols.SignedHttpRequest
+{
+    /// <summary>
+    /// A delegate that will be called to create and add additional claim(s), if set.
+    /// </summary>
+    /// <param name="payload">A SignedHttpRequest payload.</param>
+    /// <param name="signedHttpRequestDescriptor">A structure for wrapping parameters needed for SignedHttpRequest creation.</param>
+    public delegate void AdditionalClaimCreator(IDictionary<string, object> payload, SignedHttpRequestDescriptor signedHttpRequestDescriptor);
+
+    /// <summary>
+    /// A delegate that will be called to create and add the nonce claim, overriding the default behavior.
+    /// </summary>
+    /// <param name="payload">A SignedHttpRequest payload.</param>
+    /// <param name="signedHttpRequestDescriptor">A structure for wrapping parameters needed for SignedHttpRequest creation.</param>
+    public delegate void CustomNonceCreator(IDictionary<string, object> payload, SignedHttpRequestDescriptor signedHttpRequestDescriptor);
+
+    /// <summary>
+    /// Defines a set of parameters that are used by a <see cref="SignedHttpRequestHandler"/> when creating a SignedHttpRequest.
+    /// </summary>
+    public class SignedHttpRequestCreationParameters
```

Comment:
Consider merging with SignedHttpRequestDescriptor

Reply:
The reason behind having two classes is a coding model. 
CreationParameters are usually static (defined once), while the Descriptor is created for each new SignedHttpRequest.
This models fits with our S2S scenarios, and outbound policies.

Please let me know if you would still prefer to see these as a single class.

Reply:
After the review the team decided not to do this.